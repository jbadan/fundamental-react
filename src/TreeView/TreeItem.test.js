import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import TreeView from './TreeView';

// mock shortid for snapshot testing
jest.mock('shortid', () => {
    let id = 1;

    return {
        generate: () => id++
    };
});

describe('<TreeView.Item />', () => {
    const treeViewItem = (
        <TreeView.Item>
            <TreeView.Row>
                <TreeView.Col>GreatGrandchild 1</TreeView.Col>
                <TreeView.Col>Data Col 2</TreeView.Col>
                <TreeView.Col>Data Col 3</TreeView.Col>
                <TreeView.Col>Data Col 4</TreeView.Col>
            </TreeView.Row>
        </TreeView.Item>
    );

    test('create tree view item component', () => {
        let component = renderer.create(treeViewItem);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the TreeItem component', () => {
            const element = mount(
                <TreeView.Item
                    data-sample='Sample' />);

            expect(
                element.find('.fd-tree__item').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
