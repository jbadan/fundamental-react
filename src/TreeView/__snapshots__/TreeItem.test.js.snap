// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TreeView.Item /> create tree view item component 1`] = `
<li
  aria-expanded={false}
  className="fd-tree__item"
  id="fd-1"
  role="treeitem"
>
  <div
    className="fd-tree__row"
  >
    <div
      className="fd-tree__col fd-tree__col--control"
    >
      GreatGrandchild 1
    </div>
    <div
      className="fd-tree__col"
    >
      Data Col 2
    </div>
    <div
      className="fd-tree__col"
    >
      Data Col 3
    </div>
    <div
      className="fd-tree__col"
    >
      Data Col 4
    </div>
  </div>
</li>
`;
